src/amo/components/OverallRating.js

```diff
@@ -42,4 +74,38 @@ export class OverallRatingBase extends React.Component {
   }
 }
 
-export default translate({ withRef: true })(OverallRatingBase);
+export const mapStateToProps = (state) => ({
+  authToken: state.auth && state.auth.token,
+});
+
+export const mapDispatchToProps = (dispatch) => ({
+  createRating({ rating, authToken, addonID, versionID }) {
+    const postData = { rating, version: versionID };
+    log.debug('about to call API with', postData);
+    return callApi(
+      {
+        endpoint: `addons/addon/${addonID}/reviews`,
+        body: postData,
+        method: 'post',
+        auth: true,
+        state: { token: authToken },
+      })
+      .then((userRating) => {
+        dispatch(setUserRating({ addonID, userRating }));
+      })
+      .catch((error) => {
+        log.error('API error:', error);
+        // TODO: dispatch an error action instead.
+        // See https://github.com/mozilla/addons-frontend/issues/1032
+        throw error;
+      });
+  },
+});
+
+export const OverallRatingWithI18n = compose(
```

Comment:
It seems a little odd to use `compose()` with only one function but I definitely considered it in the past. It sure makes it read more clearly though, I like it.